Civi/Core/Format.php

```diff
@@ -33,7 +33,10 @@ class Format {
    * @noinspection PhpDocMissingThrowsInspection
    * @noinspection PhpUnhandledExceptionInspection
    */
-  public function money(string $amount, ?string $currency = NULL, ?string $locale = NULL): string {
+  public function money($amount, ?string $currency = NULL, ?string $locale = NULL): string {
+    if (is_null($amount) || $amount === '') {
```

Comment:
what about is (strlen($amount) === 0) ?

that would get FALSE too

Reply:
Why would the value ever be `FALSE`? And if it could be a bool, what on earth would `TRUE` mean? How much money is `TRUE` worth?

Reply:
@colemanw I doubt it would ever be TRUE but we DO sometimes fill our arrays with FALSE where we want to avoid enotices

Reply:
I guess I'm wary of type coercion, passing numbers into `strlen()` sounds like an invitation for weird bugs. I'd rather just add ` || is_bool($amount)`.

Reply:
The monetary value of TRUE is USD $1 - I think it's in the Geneva Convention. Seriously though I'd argue it should throw an error if it TRUE is passed in.

Between this and the other PR, I get the tradeoff between being able to catch bugs/typos early (being strict) and between not having to track down every existing caller (being forgiving), and normally I like the former, but my 2c on this one is to prefer this PR.

Regarding what the `if` should look like, maybe something like:
```php
// Handle values that Brick Money doesn't like
// We also treat FALSE as '' because sometimes variables are mass-filled with FALSE to avoid being missing.
if (is_null($amount) || $amount === '' || $amount === FALSE) {
  return '';
}
elseif ($amount === TRUE) {
  throw new CRM_Core_Exception('TRUE is not a valid a value for type money');
}
```

Reply:
Ok I've updated this PR to forgive `FALSE`, `NULL` and `''` but throw an exception for all other non-numeric values.

Reply:
I'm happy with that!